src/SymbolProvider/StructuredDebugDirectorySymbolProvider.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "SymbolProvider/StructuredDebugDirectorySymbolProvider.h"
+
+#include <absl/strings/str_format.h>
+
+#include <string_view>
+
+#include "OrbitBase/File.h"
+#include "OrbitBase/Result.h"
+#include "SymbolProvider/SymbolLoadingOutcome.h"
+
+namespace orbit_symbol_provider {
+
+orbit_base::Future<SymbolLoadingOutcome> StructuredDebugDirectorySymbolProvider::RetrieveSymbols(
+    const ModuleIdentifier& module_id, orbit_base::StopToken /*stop_token*/) const {
+  return {FindSymbolFile(module_id.build_id)};
+}
+
+[[nodiscard]] SymbolLoadingOutcome StructuredDebugDirectorySymbolProvider::FindSymbolFile(
```

Comment:
Very personal preference. Why not add this logic in `RetrieveSymbols` and also add a function comment in the header ?

Reply:
The main reason is that I wanted to have different return types. Since `SymbolLoadingOutcome` is secrectly an `ErrorMessageOr<>` type, I can use `OUTCOME_TRY` in `FindSymbolsFile`. 

Its in a way separating the responsibilities of the functions. Now FindSymbolsFile is responsible for finding the file and does not care about Future or being asyncronous. RetrieveSymbols uses the FindSymbolsFile and adds on top the Future handling. 

I am not entirely sure on which level we want to handle asynchronicity with SymbolProvider, but one option would be to pass a thread pool in with the constructor of the `StructuredDebugDirectorySymbolProvider` and then I only need to change the `RetrieveSymbols` method and do not need to touch `FindSymbolsFile`

Reply:
Make sense! Thanks for the explanation!